packages/core/src/shared/vscode/env.ts

```diff
@@ -94,6 +95,31 @@ export function isRemoteWorkspace(): boolean {
     return vscode.env.remoteName === 'ssh-remote'
 }
 
+/**
+ * There is Amazon Linux 2, but additionally an Amazon Linux 2 Internal.
+ * The internal version is for Amazon employees only. And this version can
+ * be used by either EC2 OR CloudDesktop. It is not exclusive to either.
+ *
+ * Use {@link isCloudDesktop()} to know if we are specifically using it.
+ *
+ * Example: `5.10.220-188.869.amzn2int.x86_64`
+ */
+export function isAmazonInternalOs() {
+    return os.version().includes('amzn2int')
+}
+
+/**
+ * Returns true if we are in an internal Amazon Cloud Desktop
+ */
+export async function isCloudDesktop() {
+    if (!isAmazonInternalOs()) {
+        return false
+    }
+
+    // This heuristic is explained in IDE-14524
+    return (await new ChildProcess('/apollo/bin/getmyfabric').run().then((r) => r.exitCode)) === 0
```

Comment:
i believe this is present on any host that can be deployed from apollo, though it would be bizarre for someone to be trying this out on a host that isn't a developer desktop

should we confirm with builder tools if this heuristic will work on the next-gen desktops?

Reply:
Thanks for bringing up. I've chatted w/ a BuilderTools member and they confirmed this won't exist in NAWS CloudDesktops. We are discussing a solution for that and can update this function once we determine one in the future.